0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+   - Any further instructions to settle should trigger payments on the shared system as normal.
+
+### Recipient collects a fee
+
+Suppose Alice and Bob's settlement arrangement is such that, before Bob can begin acknowledging incoming settlements from Alice, Bob must collect some initiation fee related to the settlement system.
+
+1. Alice sends a custom instruction to her settlement engine to settle the fee. Note that she does not use the instruction reserved for the accounting system.
+2. Alice's settlement engine automatically negotiates the fee with Bob's settlement engine (likely subject to some limit).
+3. Alice's settlement engine sends a payment on the shared ledger or system to Bob's settlement engine for the value of the fee.
+4. Bob's settlement engine, instead of informing the accounting system of the incoming settlement, does nothing, but internally tracks that the negotiated fee was paid.
+5. If Alice's accounting system triggers any subsequent settlements to Bob, Bob's settlement engine should acknowledge them to the accounting system as normal.
+
+## Specification
+
+Operating a settlement engine requires three components each exposing a particular HTTP API: a settlement engine, an accounting system, and a transport.
+
+API endpoints exposed by these systems may be classified in two ways:
+
+1. Endpoints that may be invoked by one of these other two components, and MUST NOT be invoked manually. At a minimum, all settlement engine implementations MUST support these endpoints.
+2. Endpoints that may be invoked manually by an operator.
+
+This document outlines the first class of endpoints: those only essential to automated interoperability between these systems, and not manual operation.
+
+### Accounts and identifiers
+
+Each account MUST be identified by a unique, URL-safe string.
+
+#### Accounts vs ledger identities
+
+The settlement engine is RECOMMENDED to be responsible for correlating an account identifier to the peer's identity on the shared ledger or settlement system, if required.
+
+For separation of concerns between the accounting system and the settlement system, the accounting system is NOT RECOMMENDED to have knowledge of the peer's identity on the shared settlement system, but rather only the account identifier described here.
+
+#### Multiple accounts
+
+Settlement engine implementations SHOULD support settling with multiple accounts.
+
+#### Correlating messages
+
+The transport MUST correlate incoming messages from the peer to the correct account identifier, and MUST correlate outgoing messages from an account identifier to the correct peer to send them to.
+
+### Units and quantities
+
+#### `Quantity` type
+
+A quantity represents an amount denominated in some unit of a particular fungible asset. In uses of this type, the asset should be implicit.
```

Comment:
is there any reason to allow a settlement engine to manage multiple accounts but not allow it to manage multiple currencies?

Reply:
It never said one instance couldn't support multiple currencies ;)

But since the account balance(s) are still denominated in a single asset (even if the type of asset is dynamic or something based on the account), wouldn't the asset still be implicit? I imagine in that case a given account on the SE would just be configured with a particular asset?